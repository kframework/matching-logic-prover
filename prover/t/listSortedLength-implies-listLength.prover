claim \implies( \and( listSortedLength( variable("H") { ArrayIntInt }
                                      , variable("X") { Int }
                                      , variable("K") { SetInt }
                                      , variable("MIN") { Int }
                                      , variable("Length") { Int }
                                      , .Patterns
                                      )
                    , .Patterns
                    )
              , \and( listLength( variable("H") { ArrayIntInt }
                                , variable("X") { Int }
                                , variable("K") { SetInt }
                                , variable("Length") { Int }
                                , .Patterns
                                )
                    , .Patterns
                    )
              )

strategy search-bound(5)

/*
strategy kt ; ( (right-unfold-Nth(0, 0) ; simplify ; direct-proof)
              | simplify ; direct-proof
              | (right-unfold-Nth(0, 1) ; simplify ; direct-proof)
              )
*/
