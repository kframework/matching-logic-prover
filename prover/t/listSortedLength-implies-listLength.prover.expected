<prover>
  <goal>
    <id>
      0
    </id>
    <active>
      true
    </active>
    <parent>
      -1
    </parent>
    <k>
      \implies ( \and ( listSortedLength ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "K" ) { SetInt } , variable ( "MIN" ) { Int } , variable ( "Length" ) { Int } , .Patterns ) , .Patterns ) , \and ( listLength ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "K" ) { SetInt } , variable ( "Length" ) { Int } , .Patterns ) , .Patterns ) )
    </k>
    <strategy>
      success
    </strategy>
    <trace>
      .
    </trace>
  </goal>
</prover>
